lib/misc.coffee

```diff
@@ -101,29 +171,52 @@ class Word
   getSelections: (filter = null)->
     # console.log "Word.getSelections", @left, @right
     tmp = (if _.isFunction(filter)
-      @left.concat(@right).filter(filter) or []
+      @_get("left").concat(@_get("right")).filter(filter) or []
     else
-      @left.concat(@right) or [])
+      @_get("left").concat(@_get("right")) or [])
     # console.log "returning", _.uniq tmp
     _.uniq tmp
 
 # A class describing a selection with a style (bold, italic, …)
-class Selection
-  # Word that is the left bound
-  @left = null
-  # Word that is the right bound
-  @right = null
-
+class Selection extends BaseClass
+  _name = "Selection"
+
+  _getModel: (Y, Operation) ->
+    if @_model == null
+      @_model = new Operation.MapManager(@).execute()
+      @_model.val("left", @left)
+      @_model.val("leftPos", @leftPos)
+      @_model.val("right", @right)
+      @_model.val("rightPos", @leftPos)
+      @_model.val("style", @style)
+
+      delete @left
+      delete @leftPos
+      delete @right
+      delete @rightPos
+      delete @style
+
+    return @_model
+
+  _setModel: (model) ->
+    delete @left
+    delete @left
+    delete @leftPos
+    delete @right
+    delete @rightPos
+    delete @style
+
+    @_model = model
   # Construct a new selection using the index of the first and last character.
   #
   # @param [Integer] start index of the first character
   # @param [Integer] end index of the last character
-  # @param [Rte] rte a rich-text editor (Rte) instance
+  # @param [Y.RichText] rt a richText editor (Y.RichText) instance
   # @param [Object] option options to pass to the constructor
   # @option option [Bool] bind whether or not to bind the selection
   #
-  constructor: (start, end, rte, options={})->
-    if not _.isUndefined(start) and not _.isUndefined(end) and not _.isUndefined(rte)
+  constructor: (start, end, richText, options={})->
+    if not _.isUndefined(start) and not _.isUndefined(end) and not _.isUndefined(richText)
```

Comment:
you can just call `richText?` Which is actually the same thing (parses to `richText != null`)